builder/virtualbox/ovf/config.go

```diff
@@ -1,141 +1,138 @@
-	GuestAdditionsSHA256 string   `mapstructure:"guest_additions_sha256"`
-	VMName               string   `mapstructure:"vm_name"`
-	ImportOpts           string   `mapstructure:"import_opts"`
-	ImportFlags          []string `mapstructure:"import_flags"`
-
-	ctx interpolate.Context
+	Checksum             string   `mapstructure:"checksum"`
+	ChecksumURL          string   `mapstructure:"checksum_url"`
+	ChecksumType         string   `mapstructure:"checksum_type"`
+	TargetPath           string   `mapstructure:"target_path"`
 }
 
-func NewConfig(raws ...interface{}) (*Config, []string, error) {
-	c := new(Config)
-	err := config.Decode(c, &config.DecodeOpts{
-		Interpolate:        true,
-		InterpolateContext: &c.ctx,
-		InterpolateFilter: &interpolate.RenderFilter{
-			Exclude: []string{
-				"boot_command",
-				"guest_additions_path",
-				"guest_additions_url",
-				"vboxmanage",
-				"vboxmanage_post",
-			},
-		},
-	}, raws...)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	// Defaults
-	if c.GuestAdditionsMode == "" {
-		c.GuestAdditionsMode = "upload"
-	}
-
-	if c.GuestAdditionsPath == "" {
-		c.GuestAdditionsPath = "VBoxGuestAdditions.iso"
-	}
-
-	if c.VMName == "" {
-		c.VMName = fmt.Sprintf(
-			"packer-%s-%d", c.PackerBuildName, interpolate.InitTime.Unix())
-	}
-
-	// Prepare the errors
-	var errs *packer.MultiError
-	errs = packer.MultiErrorAppend(errs, c.ExportConfig.Prepare(&c.ctx)...)
-	errs = packer.MultiErrorAppend(errs, c.ExportOpts.Prepare(&c.ctx)...)
-	errs = packer.MultiErrorAppend(errs, c.FloppyConfig.Prepare(&c.ctx)...)
-	errs = packer.MultiErrorAppend(errs, c.HTTPConfig.Prepare(&c.ctx)...)
-	errs = packer.MultiErrorAppend(errs, c.OutputConfig.Prepare(&c.ctx, &c.PackerConfig)...)
-	errs = packer.MultiErrorAppend(errs, c.RunConfig.Prepare(&c.ctx)...)
-	errs = packer.MultiErrorAppend(errs, c.ShutdownConfig.Prepare(&c.ctx)...)
-	errs = packer.MultiErrorAppend(errs, c.SSHConfig.Prepare(&c.ctx)...)
-	errs = packer.MultiErrorAppend(errs, c.VBoxManageConfig.Prepare(&c.ctx)...)
-	errs = packer.MultiErrorAppend(errs, c.VBoxManagePostConfig.Prepare(&c.ctx)...)
-	errs = packer.MultiErrorAppend(errs, c.VBoxVersionConfig.Prepare(&c.ctx)...)
+func (c * OVFConfig) Prepare(ctx *interpolate.Context) ([]string, []error) {
+	// Validation
+	var errs []error
+	var err error
+	var warnings []string
 
 	if c.SourcePath == "" {
-		errs = packer.MultiErrorAppend(errs, fmt.Errorf("source_path is required"))
-	} else {
-		if _, err := os.Stat(c.SourcePath); err != nil {
-			errs = packer.MultiErrorAppend(errs,
-				fmt.Errorf("source_path is invalid: %s", err))
-		}
+		errs = append(errs, errors.New("The source_path must be specified."))
 	}
 
-	validMode := false
-	validModes := []string{
-		vboxcommon.GuestAdditionsModeDisable,
-		vboxcommon.GuestAdditionsModeAttach,
-		vboxcommon.GuestAdditionsModeUpload,
-	}
-
-	for _, mode := range validModes {
-		if c.GuestAdditionsMode == mode {
-			validMode = true
-			break
+	if c.ChecksumType == "" {
```

Comment:
The checksum logic is not identical, but it's pretty close to verbatim of the iso_config.go